edgexfoundry/edgex-go
# refactor: Consume ProvisionWatcher DTO change for ServiceName

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  Swagger updated. 
  PW docs issues created https://github.com/edgexfoundry/edgex-docs/issues/1062

## Testing Instructions
From this branch run `make dmetadata` to build local Core Metadata docker image
From compose builder run 
- `make pull`
- `make gen no-secty
From compose builder modify the generated compose file to change core-metadata docker image to be `edgexfoundry/core-metadata:0.0.0-dev`
From compose builder run `make up`
Send REST command `http://localhost:59881/api/v3/provisionwatcher/name/Simple-Provision-Watcher`
Verify response is not found 
Send REST command http://localhost:59881/api/v3/provisionwatcher/service/name/device-simple
Verify response is empty list 
Build Device Simple using branch from this PR: https://github.com/edgexfoundry/device-sdk-go/pull/1453
Run Device Simple from command line with `-cp -d -i ` flags
Verify from logs that it didn't find Simple-Provision-Watcher and added it
Send REST command `http://localhost:59881/api/v3/provisionwatcher/name/Simple-Provision-Watcher`
Verify response is the Simple-Provision-Watcher
Send REST command http://localhost:59881/api/v3/provisionwatcher/service/name/device-simple
Verify response contains Simple-Provision-Watcher 
Restart Device Simple
Verify from logs that it found Simple-Provision-Watcher and is using it.

## New Dependency Instructions (If applicable)
